### DISCUSSION TITLE
Questions about adding datasets into EL1000

### DISCUSSION THREAD

**Discussion by marianne-m** [LAAC-LSCP/ChildProject#312] [Q&A] [answered]
Hi ! I'm following the [instructions](https://gin.g-node.org/EL1000/EL1000/issues/3) for uploading a new dataset into EL1000, and I have two questions/remarks on STEP TWO.

1/ The script [metadata.py](https://gin.g-node.org/EL1000/lyon/src/main/scripts/metadata.py) is used, with the following importation : 
 `from EL1000.metadata import MetadataImporter`

However there is no documentation on this. 

So I installed EL1000/tools : 
```
git clone git@gin.g-node.org:/EL1000/tools.git
cd tools
pip install .
```

2/ This same script [metadata.py](https://gin.g-node.org/EL1000/lyon/src/main/scripts/metadata.py) is called in [```creation.sh```](https://gin.g-node.org/EL1000/rague/src/main/scripts/creation.sh) with this line :
```python scripts/metadata.py rague metadata/confidential/original/purdue_ndd_metadata.csv --recompute-ages ```

But ```scripts/metadata.py``` doesn't take any arguments. 
Is it the right script ? or is it because the `metadata.py` script used to take arguments ?

**Comment by lucasgautheron** [comment, marked as answer]
Hello Marianne!

1/ What you did was right! Sorry about the lack of documentation; initially I introduced the EL1k package to speed up my own work... It's time I document it properly though!

2/ I think creation.sh was written by Alex before there was the EL1k package; so yeah, it used to take arguments, but it no longer does.

The arguments are passed to the MetadataImporter constructor now: https://gin.g-node.org/EL1000/lyon/src/main/scripts/metadata.py#L14

Here is more information about the different options: 

https://gin.g-node.org/EL1000/tools/src/master/EL1000/metadata.py#L28-L51

**Reply by marianne-m** [reply]
Okay, thank you !
